Assets/HoloToolkit/Utilities/Scripts/CanvasRaycastResult.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace HoloToolkit.Unity
+{
+    public struct CanvasRaycastResult : IComparable<CanvasRaycastResult>
```

Comment:
We should probably keep this inside the `GazeManager` class as a private struct, or at the very least move this class into the input feature area scripts/utilities folder.

Reply:
Instead, I could also move all of this logic out of the GazeManager and add an extension as `EventSystem.current.Raycast(UnityUIPointerEvent, RaycastLayerMasks)`. This solves a general problem with Unity until they finally fix their `RaycastAll` method to sort by separate canvases first.

Reply:
That also sounds like a great idea. I'd just like to keep all the feature area's scripts together in their respective folders.

I'm slowly trying to decouple the feature areas so they can work on their own if needed.